src/initialize.F90

```diff
@@ -902,4 +916,230 @@ subroutine allocate_banks()
+
+    integer, intent(out), allocatable     :: univ_list(:) ! Target offsets
+    integer, intent(out), allocatable     :: kounts(:,:)  ! Target count
+    logical, intent(out), allocatable     :: found(:,:)   ! Target found
+
+    integer :: i, j, k, l, m                    ! Loop counters
+    type(DictIntInt) :: cell_list               ! distribells to track    
+    type(Universe),    pointer :: univ          ! pointer to universe
+    class(Lattice),    pointer :: lat           ! pointer to lattice
+    type(TallyObject), pointer :: tally         ! pointer to tally
+    type(TallyFilter), pointer :: filter        ! pointer to filter
+    
+    ! Begin gathering list of cells in distribcell tallies
+    n_maps = 0
+    
+    ! Populate list of distribcells to track
+    do i = 1, n_tallies
+      tally => tallies(i)
+
+      do j = 1, tally % n_filters
+        filter => tally % filters(j)        
+
+        if (filter % type == FILTER_DISTRIBCELL) then
+          if (.not. cell_list % has_key(filter % int_bins(1))) then
+            call cell_list % add_key(filter % int_bins(1),0)
+          end if
+        end if 
+
+      end do
+    end do
+    
+    ! Compute the number of unique universes containing these distribcells
+    ! to determine the number of offset tables to allocate
+    do i = 1, n_universes
+      univ => universes(i)
+
+      do j = 1, univ % n_cells
+
+        if (cell_list % has_key(univ % cells(j))) then
+          n_maps = n_maps + 1
+          cycle
+        end if
+
+      end do
+    end do
+    
+    ! Allocate the list of offset tables for each unique universe
+    allocate(univ_list(n_maps))
+
+    ! Allocate list to accumulate target distribccell counts in each universe
+    allocate(kounts(n_universes, n_maps))
+
+    ! Allocate list to track if target distribcells are found in each universe
+    allocate(found(n_universes, n_maps))
+
+    do i = 1, n_universes
+      do j = 1, n_maps
+          kounts(i,j) = 0
+          found(i,j) = .false.
+        end do
+    end do
+
+    k = 1
+    do i = 1, n_universes
+    
+      univ => universes(i)
+
+
+      do j = 1, univ % n_cells
+      
+        if (cell_list % has_key(univ % cells(j))) then
+          
+            ! Loop over all tallies    
+            do l = 1, n_tallies
+              tally => tallies(l)
+              
+              do m = 1, tally % n_filters
+                filter => tally % filters(m)
+                
+                ! Loop over only distribcell filters
+                ! If filter points to cell we just found, set offset index
+                if (filter % type == FILTER_DISTRIBCELL) then                  
+                  if (filter % int_bins(1) == univ % cells(j)) then
+                    filter % offset = k
+                  end if
+                end if
+
+              end do
+            end do          
+          
+          univ_list(k) = univ % id
+          k = k + 1
+          cycle
```

Comment:
Good call, this is a residual from the implementation before I restructured - I just removed it.